### PR TITLE
store info from organ registration form on the user

### LINKED ISSUE #6557: Store user fields after a user submits organ reg form
We should be able to fill out on a user once they submit the organ donation form
- [ ] `field_birthdate`
- [ ] `field_mobile`
- [ ] `field_address`
- [ ] `field_last_name`

### PR DESCRIPTION
**Pull request by katiecrane**
#### What's this PR do?

Updates user name, address, and birthday with the information from the organ donation registration form.
#### How should this be reviewed?

Register to give us your organs and change some of your data. Then check to make sure it updated.
#### Relevant tickets

Fixes #6557

### CODE REVIEW

File: lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module [hunk 1]
```diff
@@ -90,6 +90,32 @@ function dosomething_organ_donation_post_registration() {
 
   $response = drupal_http_request($client['base_url'] . '/v2/registrations/', $options);
 
+  // Load the user so we can make updates
+  global $user;
+  $account = user_load($user->uid);
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module]: `global $user` should be the full loaded object, you shouldn't need to re-load here

File: lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module [hunk 2]
```diff
@@ -90,6 +90,32 @@ function dosomething_organ_donation_post_registration() {
...
+  $account->field_address[LANGUAGE_NONE][0] = $new_field_address;
+
+  // Update name and birthday
+  $new_fields = [
+    'first_name' => $params['first_name'],
+    'last_name' => $params['last_name'],
+    'birthdate' => date('Y-m-d', strtotime($params['birthdate'])),
+  ];
+  $formatted_fields = [];
+  dosomething_user_set_fields($formatted_fields, $new_fields);
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module]: this function works if you pass an empty array?

File: lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module [hunk 3]
```diff
@@ -90,6 +90,32 @@ function dosomething_organ_donation_post_registration() {
 
   $response = drupal_http_request($client['base_url'] . '/v2/registrations/', $options);
 
+  // Load the user so we can make updates
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module]: we should break this whole chunk out into it's own function

File: lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module (hunk 1 above)
Comment (katiecrane) [inline lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module]: aha! you're right, something else must have fixed the problem that caused me to add this

File: lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module (hunk 2 above)
Comment (katiecrane) [inline lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module]: @angaither it sure does. I got it to work but I'm still confused by that function

Comment (angaither) [inline lib/modules/dosomething/dosomething_organ_donation/dosomething_organ_donation.module]: understandable

**Comment by angaither** [comment]
lookin 🚢 able

**Pull request merged**